scripts/aad-app-reg.sh

```diff
@@ -379,6 +337,7 @@ JSON
 
 declare roleUserReadAll=$(get_msgraph_role "User.Read.All" )
 declare roleDirectoryReadAll=$(get_msgraph_role "Directory.Read.All" )
+declare roleApplicationReadWriteAll=$(get_msgraph_role "Application.ReadWrite.All" )
```

Comment:
Do we need `ReadWrite.All` ? If it's so that this app can create / manage others - I think we'd get away with `Application.ReadWrite.OwnedBy` - bit more tightly scoped to the apps for TRE? Definitions here - https://docs.microsoft.com/en-us/graph/permissions-reference#application-permissions-4

Reply:
With OwnedBy
{"error":{"code":"Authorization_RequestDenied","message":"Insufficient privileges to complete the operation.","innerError":{"date":"2022-03-28T08:33:23","request-id":"4bd4caa4-bc35-4b5e-90ff-ee17efdcfb67","client-request-id":"4bd4caa4-bc35-4b5e-90ff-ee17efdcfb67"}}}

Reply:
Put it back to "All" and it works again. :-( I am unsure why - happy to walk through it as may have missed something - could put a note in the code?

Reply:
yea it's not a massive deal - the app is definitely an owner of the one it's trying to modify? but if not let's use `.All` then.

Reply:
I agree that it would be preferable to use `Application.ReadWrite.OwnedBy`.
Sounds like the app isn't the owner of the object it's trying to modify? If so, is that something that we expect will change as this work progresses? (i.e. add a comment in the code and an issue to track re-visiting this once the work is further on)